render/wgpu/src/pipelines.rs

```diff
@@ -1,10 +1,11 @@
 use crate::{Error, MaskState, Vertex};
-use enum_map::{enum_map, EnumMap};
+use enum_map::{Enum, EnumMap};
+use ruffle_core::swf::BlendMode;
 use wgpu::vertex_attr_array;
 
 #[derive(Debug)]
 pub struct ShapePipeline {
-    pub mask_pipelines: EnumMap<MaskState, wgpu::RenderPipeline>,
+    pub pipelines: EnumMap<BlendMode, EnumMap<MaskState, wgpu::RenderPipeline>>,
```

Comment:
Instead of a nested `EnumMap`, we could use a single layer hash of the states, something like:

```
type PipelineState = (BlendMode, MaskState);
pub pipelines:  FnvHashMap<PipelineState, wgpu::RenderPipeline>;
```

Reply:
Actually scratch that; I was thinking `EnumMap` did a hash anyway, but it's simply a flat array, so let's avoid hashing for now. It still might be nice to drop the `enum_map` dependency in any case and just use an array here. Could be a future PR.